packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingModule.java

```diff
@@ -216,18 +218,19 @@ public void onNewIntent(Intent intent) {
       if (messageId == null) messageId = intent.getExtras().getString("message_id");
 
       if (messageId != null) {
+        ReactNativeFirebaseEventEmitter emitter = ReactNativeFirebaseEventEmitter.getSharedInstance();
         RemoteMessage remoteMessage = ReactNativeFirebaseMessagingReceiver.notifications.get(messageId);
-        if (remoteMessage == null) {
-          remoteMessage = popRemoteMessageFromMessagingStore(messageId);
-        }
 
-        if (remoteMessage != null) {
-          initialNotification = remoteMessage;
+        if (remoteMessage == null) {
+          WritableMap remoteMessageMap = popRemoteMessageMapFromMessagingStore(messageId);
+          initialNotification = remoteMessageMap;
+          emitter.sendEvent(ReactNativeFirebaseMessagingSerializer.remoteMessageMapToEvent(remoteMessageMap, true));
+        } else {
+          initialNotification = ReactNativeFirebaseMessagingSerializer.remoteMessageToWritableMap(remoteMessage);
           ReactNativeFirebaseMessagingReceiver.notifications.remove(messageId);
-
-          ReactNativeFirebaseEventEmitter emitter = ReactNativeFirebaseEventEmitter.getSharedInstance();
           emitter.sendEvent(ReactNativeFirebaseMessagingSerializer.remoteMessageToEvent(remoteMessage, true));
         }
+        ReactNativeFirebaseMessagingReceiver.notifications.remove(messageId);
```

Comment:
similar to above, you've changed the logical structure *and* the data types at the same time, which makes this difficult to untangle while reading it.

It appears that all you really *must* do is pop the RemoteMessageMap vs RemoteMessage ?

Can you do this chunk with the same effect but *only* altering the data types to be the Map you need?

Reply:
Same here, I made the most minimal changes I could think of.